### PR TITLE
fix the Ingress apiVersion so it isn't a deprecated version

### PR DESCRIPTION
**Pull request by jmazzitelli**
fixes: https://github.com/kiali/kiali/issues/3706

### CODE REVIEW

**Comment by jmazzitelli** [comment]
merge this first: https://github.com/kiali/kiali-operator/pull/251 (this PR is built on top of that one)

**Comment by jmazzitelli** [comment]
Hmm... this v1 version does not look to be available in k8s versions before 1.19. Before merging, need to figure out if this is a problem. I suspect it is - I think k8s versions 1.18 and lower might still be used and not EOL'ed.

**Comment by jmazzitelli** [comment]
Moved to draft - only merge this when we know we only support k8s environments at the 1.19 version and up.

**Comment by jmazzitelli** [comment]
don't forget to change some hack scripts that use `extensions/v1beta1` - change to `networking.k8s.io/v1`

https://github.com/kiali/kiali/search?q=extensions%2Fv1beta1

In addition, anything using https://raw.githubusercontent.com/maistra/istio/maistra-2.0/samples/bookinfo/platform/kube/bookinfo-ingress.yaml needs to change because it, too, uses the deprecated version.

https://github.com/kiali/kiali/search?q=bookinfo-ingress

**Comment by jmazzitelli** [comment]
molecule tests pass

File: molecule/instance-name-test/converge.yml
Comment (nrfox) [inline molecule/instance-name-test/converge.yml:63]: This keeps backwards compatibility with 1.18 for the time being as v1beta1 is the fallback right?

Comment (jmazzitelli) [reply molecule/instance-name-test/converge.yml:63]: correct... if that lookup failed to find "v1" that if-stmt will fallback to v1beta1 (which is what we used before - so this is backwards compatible).

**Review by nrfox — APPROVED** [review]
LGTM and works on my setup

**Review by pbajjuri20 — APPROVED** [review]
I run the regression test, everything works perfectly on OpenShift. LGTM

**Pull request merged**
